cartography/intel/oci/iam.py

```diff
@@ -0,0 +1,312 @@
+#Copyright (c) 2020, Oracle and/or its affiliates.
+# OCI Identity API-centric functions
+# https://docs.cloud.oracle.com/iaas/Content/Identity/Concepts/overview.htm
+import oci
+from . import utils
+import json
+import re
+import logging
+from collections import namedtuple
+from string import Template
+
+from cartography.util import run_cleanup_job
+
+logger = logging.getLogger(__name__)
+
+def sync_compartments(neo4j_session, iam, current_tenancy_id, oci_update_tag, common_job_parameters):
+    logger.debug("Syncing IAM compartments for account '%s'.", current_tenancy_id)
+    data = get_compartment_list_data(iam,current_tenancy_id)
+    load_compartments(neo4j_session, data['Compartments'], current_tenancy_id, oci_update_tag)
+    run_cleanup_job('oci_import_compartments_cleanup.json', neo4j_session, common_job_parameters)
+
+def get_compartment_list_data_recurse(iam, compartment_list, compartment_id):
+    response = oci.pagination.list_call_get_all_results(iam.list_compartments,compartment_id)
+    if not response.data:
+        return
+    compartment_list.update({"Compartments":list(compartment_list["Compartments"])+utils.oci_object_to_json(response.data)})
+    for compartment in response.data:
+        get_compartment_list_data_recurse(iam, compartment_list, compartment.id)
+
+def get_compartment_list_data(iam,current_tenancy_id):
+    compartment_list = {"Compartments": ""}
+    get_compartment_list_data_recurse(iam, compartment_list, current_tenancy_id)
+    return compartment_list
+
+def load_compartments(neo4j_session, compartments, current_oci_tenancy_id, oci_update_tag):
+    ingest_compartment = """
+    MERGE (cnode:OCICompartment{ocid: {OCID}})
+    ON CREATE SET cnode:OCICompartment, cnode.firstseen = timestamp(),
+    cnode.createdate = {CREATE_DATE}
+    SET cnode.name = {NAME}, cnode.compartmentid = {COMPARTMENT_ID}
+    WITH cnode
+    MATCH (aa) WHERE (aa:OCITenancy OR aa:OCICompartment) AND aa.ocid={COMPARTMENT_ID}
```

Comment:
+1. Try to make an inheritance/subclass-like structure by adding a `:GenericLabel` and a `:SpecificLabel`. When the cleanup job runs we can do something like a  `detach delete` on the `:GenericLabel` and it will cover the `:SpecificLabels` as well.